src/Apps/Collect/urlBuilder.ts

```diff
@@ -0,0 +1,64 @@
+import qs from "qs"
+import { State } from "./FilterState"
+
+export const buildUrlForCollectionApp = (state: State): string => {
+  const params = removeDefaultValues(state)
+  const queryString = qs.stringify(params)
+
+  return queryString
+    ? `${window.location.pathname}?${queryString}`
+    : window.location.pathname
+}
+
+export const buildUrlForCollectApp = (state: State): string => {
+  const fragment = buildCollectUrlFragmentFromState(state)
+
+  return fragment ? `/collect${fragment}` : "/collect"
+}
+
+const buildCollectUrlFragmentFromState = (state: State): string => {
+  const { medium, ...params } = removeDefaultValues(state)
+
+  const emptyOrSpecificMedium = medium ? `/${medium}` : ""
+
+  if (Object.keys(params).length === 0) {
+    return emptyOrSpecificMedium
+  }
+
+  return `${emptyOrSpecificMedium}?${qs.stringify(params)}`
+}
+
+const removeDefaultValues = (state: State): State => {
+  return Object.entries(state).reduce((acc, [key, value]) => {
+    if (isDefaultFilter(key, value)) {
+      return acc
+    } else {
+      return { ...acc, [key]: value }
+    }
+  }, {})
+}
+
+// This is used to remove default state params that clutter up URLs.
+const isDefaultFilter = (filter, value): boolean => {
```

Comment:
I'm going to add some tests to cover this stuff. I don't believe it's currently covered.

Reply:
Tests added. I also refactored the default value checks to be a `switch` instead of an `if`, as had been suggested several times 😄